File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end());
        int l = points.size();
        int c = 1;
        int pv = points[0][1];
        for (int i = 1; i < l; i++) {
            if (points[i][0] > pv) {
                c++;
                pv = points[i][1];
            }
            else{
                pv = min(pv,points[i][1]);
            }
        }
        return c;
    }
};
// class Solution {
// public:
//     int findMinArrowShots(vector<vector<int>>& points) {
//         if (points.empty()) {
//             return 0;
//         }

//         sort(points.begin(), points.end(), [](const vector<int>& a, const vector<int>& b) {
//             return a[1] < b[1];
//         });

//         int arrows = 1;
//         int end = points[0][1];

//         for (const vector<int>& interval : points) {
//             if (interval[0] > end) {
//                 arrows++;
//                 end = interval[1];
//             }
//         }

//         return arrows;
//     }
// };

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        //sort based on X0
        
        Arrays.sort(points, (a,b) -> Integer.compare(a[1], b[1]));
        
        // for(int i = 0; i<points.length; i++){
        //     System.out.println(points[i][0] + " " + points[i][1]);
        // }
        
        
        int endPnt = points[0][1];
        int cnt = 1;
        
        for(int i = 1; i<points.length; i++){
            if(points[i][0] > endPnt){
                endPnt = points[i][1];
                cnt++;
            }
        }
        
        return cnt;
    }
}